### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by kingdonb**
Working on upgrades for the next patch release

### CODE REVIEW

**Comment by kingdonb** [comment]
I lined up all the release PRs, so when this one gets approved I should be able to knock them all down and release in about an hour 👍

**Comment by kingdonb** [comment]
WDYT @pjbgf ? `kustomize` is pinned/frozen in place at 3.8.4 and `sops` is already at 3.7.3, the latest. I think that's everything, except for Kubernetes. Let me know if you find anything I have missed, I'll add it.

Do you think there are any other version pins we should advance?

K8s 1.25.x is out now, which means K8s 1.21.x series is finally out of support (EOL). We have been pinning this series (1.21.x) and planned on upgrading past the breaking issues from those dependencies again around this milestone, about when there should be no more people still running on K8s 1.16.x, I have lost track of the details about supporting at the edges. I thought we had an issue pinned where we discussed how we plan to handle this, but I'm not seeing it. Will search for it again later.

I think that we might still want to consider publishing that type of upgrade but it should come later, if at all, in a next MINOR v1.26.0. I am not aware of any issues with K8s 1.24 or 1.25 preventing Flux v1 users from advancing on that upgrade path. We can release these upgrades as a patch now, and plan those upgrades for next release. That's what I'd propose.

That being said, I have not yet tested against K8s 1.25.0, I do not have a test matrix of different Kubernetes release versions at my disposal for testing against, as my maintainer duties are not resourced in that way here. I have plenty of ARM64 resources but Flux v1 only runs on AMD64 officially. I will try to make a 1.25.0 cluster and root out any issues with the e2e.

---

We are not expecting to make major changes in the Flux v1 codebase again. I think there are some deprecation warnings that may impact users in the strange place of being on the very latest Kubernetes release but also pinned to Flux legacy v1, there are probably no users in this position. (It seems much more likely that Flux v1 runs on systems which are never upgraded.)

I have not checked the deprecation schedule for this release out yet, but I think we are due for another breaking change requiring upgrade soon from upstream Kubernetes. Maybe someone else is more on top of this information than I am today.

File: go.mod [hunk 1]
```diff
@@ -22,6 +22,9 @@ replace (
 	k8s.io/code-generator => k8s.io/code-generator v0.21.14
 )
 
+// Versions v3.0.0>x>v2.4.0 are susceptible to CVE-2022-28948
+replace gopkg.in/yaml.v3 => gopkg.in/yaml.v3 v3.0.1
+
```
Comment (pjbgf) [inline go.mod:27]:
```suggestion

// Fix CVE-2022-1996
replace github.com/emicklei/go-restful => github.com/emicklei/go-restful v2.16.0+incompatible

```

**Review by pjbgf — CHANGES_REQUESTED** [review]
From my checks, we can fix an additional CVE as per below.

**Comment by pjbgf** [comment]
@kingdonb I agree with the approach, let's take the Kubernetes update as a separate thread/release. In the coming days I plan to take a look around Flux 1 and the impact of 1.21.x being EOL and [1.22.x becoming EOL soon](https://kubernetes.io/releases/#release-v1-22).

**Comment by kingdonb** [comment]
I think #3564 is the issue that I was looking for. If there are any more deprecation warnings to address, let's open an issue.

Reopened:
* #3564

**Review by pjbgf — APPROVED** [review]
LGTM

Thanks @kingdonb for working on this. :bow:

**Pull request merged**